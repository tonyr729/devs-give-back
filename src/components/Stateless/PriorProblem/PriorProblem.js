import React from 'react';
import { NavLink } from 'react-router-dom';
import './PriorProblem.css'

export const PriorProblem = () => {
  return (
    <div>
      <p className='sorry-text'>Sorry,</p>
      <p className='info-message'>You can only have one listed problem at a time.</p>
      <NavLink to='/client-profile'>
        <p>Go to my profile</p>
      </NavLink>
    </div>
  );
};