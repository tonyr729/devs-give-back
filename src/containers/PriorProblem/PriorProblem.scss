.sorry-text {
  font-size: 2rem;
  padding-top: 10rem;
}

.info-message {
  font-size: 4rem;
  padding: 4rem 0 8rem;
  width: 60rem;
  line-height: 5rem;
}

.nav-button {
  background: none;
  border: none;
  color: #f8f8ff;
  width: 16rem;
  height: 4rem;
  margin: 0 auto;
  display: block;
  font-size: 1.3rem;
  &:hover {
    color: #29bbff;
  }
}