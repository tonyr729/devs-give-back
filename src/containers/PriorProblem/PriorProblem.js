import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import ApiHelper from '../../helpers/apiCalls';
import { connect } from 'react-redux';
import {createProblemTitle, createProblemBody, createProblemClient } from '../../actions/actions';
import './PriorProblem.css'

export class PriorProblem extends Component {
  constructor() {
    super();
    this.state = {
      problem: {}
    }
    this.api = new ApiHelper();
  }

  componentDidMount = async () => {
    if (this.props.client) {
      const clientID = this.props.client.id 
      const problem = await this.api.findMatchingProblem(clientID);
      this.setState({
        problem: problem
      })
    }
  }
  
  handleClick = () => {
    if (this.state.problem){
      this.props.createProblemClient(this.state.problem.clientID)
      this.props.createProblemTitle(this.state.problem.title)
      this.props.createProblemBody(this.state.problem.body)
      this.props.history.push("/client-profile");
    } else {
      this.props.history.push("/client-login")
    }
  }

  render() {
    return (
      <div className='client-background'>
        <p className='sorry-text'>Sorry,</p>
        <p className='info-message'>You can only have one listed problem at a time.</p>
          <button className='nav-button' onClick={ this.handleClick }>Go to my profile</button>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  client: state.client 
});

const mapDispatchToProps = (dispatch) => ({
  createProblemClient: (client) => dispatch(createProblemClient(client)),
  createProblemTitle: (title) => dispatch(createProblemTitle(title)),
  createProblemBody: (body) => dispatch(createProblemBody(body))
});

export default connect(mapStateToProps, mapDispatchToProps)(PriorProblem);