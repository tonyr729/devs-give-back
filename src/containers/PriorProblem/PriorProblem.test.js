import { PriorProblem } from './PriorProblem';
import React from 'react';
import { shallow } from 'enzyme';

describe('PriorProblem', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<PriorProblem />);

    expect(wrapper).toMatchSnapshot();
  });

  
});