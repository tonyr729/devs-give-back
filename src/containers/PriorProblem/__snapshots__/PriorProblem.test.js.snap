// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriorProblem should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PriorProblem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="sorry-text"
>
          Sorry,
</p>,
        <p
          className="info-message"
>
          You can only have one listed problem at a time.
</p>,
        <button
          className="nav-button"
          onClick={[Function]}
>
          Go to my profile
</button>,
      ],
      "className": "client-background",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry,",
          "className": "sorry-text",
        },
        "ref": null,
        "rendered": "Sorry,",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You can only have one listed problem at a time.",
          "className": "info-message",
        },
        "ref": null,
        "rendered": "You can only have one listed problem at a time.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Go to my profile",
          "className": "nav-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Go to my profile",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="sorry-text"
>
            Sorry,
</p>,
          <p
            className="info-message"
>
            You can only have one listed problem at a time.
</p>,
          <button
            className="nav-button"
            onClick={[Function]}
>
            Go to my profile
</button>,
        ],
        "className": "client-background",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry,",
            "className": "sorry-text",
          },
          "ref": null,
          "rendered": "Sorry,",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You can only have one listed problem at a time.",
            "className": "info-message",
          },
          "ref": null,
          "rendered": "You can only have one listed problem at a time.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go to my profile",
            "className": "nav-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Go to my profile",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
